### PR TITLE
Feat toggle message is read

### PR DESCRIPTION
**Pull request by subject026**
Can now send put request to toggle message `isRead` field.

### CODE REVIEW

File: src/message/message.service.ts [hunk 1]
```diff
@@ -27,11 +27,20 @@ export class MessageService {
...
 
-  update(id: number, updateMessageDto: UpdateMessageDto) {
-    return `This action updates a #${id} message`;
-  }
+  async updateById(id: number, updateMessageDto: UpdateMessageDto) {
+    const message = await this.messageRepository.findOneBy({ id });
+
+    if (!message)
+      throw new HttpException(
+        'Message not found, cannot create note.',
```
Comment (DavidRecheni) [inline src/message/message.service.ts:35]: Will this make the backend crash? I think we maybe should return a 404 : Message not found

Comment (subject026) [reply src/message/message.service.ts:35]: This gets caught and sent as the response. 400 might not be the right code though, either 400 or 404....

I suppose the request was valid but a message can't be found with the id given. Probably 404? 

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses

Comment (subject026) [reply src/message/message.service.ts:35]: I've added a test for this now

Comment (subject026) [reply src/message/message.service.ts:35]: Actually I think it should be 400.

If the request is to create a note and the message id provided doesn't exist in the db then it's a bad request surely?

Comment (DavidRecheni) [reply src/message/message.service.ts:35]: Yeah I think both are valid

**Pull request merged**